eZ/Bundle/EzPublishCoreBundle/Imagine/Cache/ResolverFactory.php

```diff
@@ -0,0 +1,84 @@
+<?php
+
+/**
+ * @copyright Copyright (C) eZ Systems AS. All rights reserved.
+ * @license For full copyright and license information view LICENSE file distributed with this source code.
+ */
+namespace eZ\Bundle\EzPublishCoreBundle\Imagine\Cache;
+
+use eZ\Publish\Core\MVC\ConfigResolverInterface;
+use Liip\ImagineBundle\Imagine\Cache\Resolver\ResolverInterface;
+
+class ResolverFactory
+{
+    /**
+     * @var \eZ\Publish\Core\MVC\ConfigResolverInterface
+     */
+    private $configResolver;
+
+    /**
+     * @var \Liip\ImagineBundle\Imagine\Cache\Resolver\ResolverInterface
+     */
+    private $resolver;
+
+    /**
+     * @var string|null
+     */
+    private $resolverDecoratorClass;
+
+    /**
+     * @var array
+     */
+    private $hosts = [];
+
+    /**
+     * @var string
+     */
+    private $proxyResolverClass;
+
+    /**
+     * @var string
+     */
+    private $relativeResolverClass;
+
+    /**
+     * @param \eZ\Publish\Core\MVC\ConfigResolverInterface $configResolver
+     * @param \Liip\ImagineBundle\Imagine\Cache\Resolver\ResolverInterface $resolver
+     * @param string $proxyResolverClass
+     * @param string $relativeResolverClass
+     */
+    public function __construct(
+        ConfigResolverInterface $configResolver,
+        ResolverInterface $resolver,
+        $proxyResolverClass,
+        $relativeResolverClass
+    ) {
+        $this->configResolver = $configResolver;
+        $this->resolver = $resolver;
+        $this->proxyResolverClass = $proxyResolverClass;
+        $this->relativeResolverClass = $relativeResolverClass;
+    }
+
+    /**
+     * @return \Liip\ImagineBundle\Imagine\Cache\Resolver\ResolverInterface
+     */
+    public function createCacheResolver()
+    {
+        if ($this->configResolver->hasParameter('image_host')
+            && ($imageHost = $this->configResolver->getParameter('image_host')) !== '') {
```

Comment:
Nitpick: Can't we set `$imageHost` before `if`? That would increase readability of the code.
```
$imageHost = $this->configResolver->hasParameter('image_host') ? $this->configResolver->getParameter('image_host') : '';
if ($imageHost !== '') {
```

Reply:
@mateuszbieniek thanks to your remark I slightly simplified `ResolverFactory`. I hope it's better now, see: https://github.com/ezsystems/ezpublish-kernel/pull/2511/commits/46ce6ebda5d54aaafae88dafe198740e9c0abbb3